gobblin-hive-registration/src/main/java/gobblin/hive/policy/HiveRegistrationPolicyBase.java

```diff
@@ -97,10 +109,40 @@ public HiveRegistrationPolicyBase(State props) {
    * This method first tries to obtain the database name from {@link #HIVE_DATABASE_NAME}.
    * If this property is not specified, it then tries to obtain the database name using
    * the first group of {@link #HIVE_DATABASE_REGEX}.
+   *
+   */
+  protected Optional<String> getDatabaseName(Path path) {
+    if (!this.props.contains(HIVE_DATABASE_NAME) && !this.props.contains(HIVE_DATABASE_REGEX)) {
+      return Optional.<String> absent();
+    }
+
+    return Optional.<String> of(
+        this.dbNamePrefix + getDatabaseOrTableName(path, HIVE_DATABASE_NAME, HIVE_DATABASE_REGEX, this.dbNamePattern)
+            + this.dbNameSuffix);
+  }
+
+  /**
+   * Obtain Hive database names. The returned {@link Iterable} contains the database name returned by
+   * {@link #getDatabaseName(Path)} (if present) plus additional database names specified in
+   * {@link #ADDITIONAL_HIVE_DATABASE_NAMES}.
+   *
    */
-  protected String getDatabaseName(Path path) {
-    return this.dbNamePrefix + getDatabaseOrTableName(path, HIVE_DATABASE_NAME, HIVE_DATABASE_REGEX, this.dbNamePattern)
-        + this.dbNameSuffix;
+  protected Iterable<String> getDatabaseNames(Path path) {
+    List<String> databaseNames = Lists.newArrayList();
+
+    Optional<String> databaseName;
+    if ((databaseName = getDatabaseName(path)).isPresent()) {
+      databaseNames.add(databaseName.get());
+    }
+
+    if (!Strings.isNullOrEmpty(this.props.getProp(ADDITIONAL_HIVE_DATABASE_NAMES))) {
+      for (String additionalDbName : this.props.getPropAsList(ADDITIONAL_HIVE_DATABASE_NAMES)) {
+        databaseNames.add(this.dbNamePrefix + additionalDbName + this.dbNameSuffix);
```

Comment:
I don't foresee a use case where one needs to use two or more regexes for db or table names. This class just provides a base implementation; someone who has a fancy use case can extend it.